### PR TITLE
DROOLS-4650 : GDST: Re-editing BRL columns causes NPE

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-4650

### CODE REVIEW

**Comment by Rikkola** [comment]
I'll try to get green builds before review. I fear this is a risky change.

**Comment by manstis** [comment]
@Rikkola I think this is also a fix for https://github.com/kiegroup/drools-wb/pull/1212?

It might be worth linking the JIRA, if that is true.

**Comment by Rikkola** [comment]
@manstis This fixes that. There are two dublicate product tickets, but only linked this to one. I'll link both.

**Comment by Rikkola** [comment]
All green, nice. I'll fix the code smell, but removing the diamond operator should not change behaviour.

**Comment by Rikkola** [comment]
@jomarko @manstis if you two would be kind enough to share some thoughts about the changes.

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/ConditionCol52.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
         HasParameterizedOperator,
         HasBinding {
 
-    private static final long serialVersionUID = 510l;
+    private static final long serialVersionUID = 729l;
```
Comment (Rikkola) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/ConditionCol52.java:37]: No idea if these matter, but thought that I'll update these to current version :)

Comment (jomarko) [reply drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/ConditionCol52.java:37]: Shouldn't we change it witch each community release then?

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/DTColumnConfig52.java [hunk 2]
```diff
@@ -171,10 +171,12 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (!(o instanceof DTColumnConfig52)) {
+
+        if(o == null || o.getClass() != this.getClass()){
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/DTColumnConfig52.java]: This change is not needed, https://docs.oracle.com/javase/specs/jls/se11/html/jls-15.html#jls-15.20.2

**Review by jomarko — CHANGES_REQUESTED** [review]
Screenshot from 2019-10-18 07-34-23

Still doesn't work for BRL action columns

**Comment by jomarko** [comment]
@Rikkola as this is about NPE in equals so I (understand my Intelij Idea) reimplemented equals of dtables, I guess also rule model need similar update. https://github.com/Rikkola/drools/pull/6

It solved issue I reported https://github.com/kiegroup/drools/pull/2587#pullrequestreview-303673306

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/ConditionCol52.java (hunk 1 above)
Comment (Rikkola) [reply drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/ConditionCol52.java:37]: Probably only worth it to update when a release has changes.

**Comment by Rikkola** [comment]
@jomarko Is this now ok from you side? Retested and can't figure out how to break it.

**Review by jomarko — APPROVED** [review]

**Comment by Rikkola** [comment]
@manstis Hi, I could use your feedback on this.

**Review by manstis — APPROVED** [review]
Thanks @Rikkola so my [suspicion](https://github.com/kiegroup/drools-wb/pull/1212#issuecomment-520838543) about equals somewhere was correct.

**Comment by Rikkola** [comment]
@manstis Yep that helped. Although debugging to figure out what exactly is the problem was PITA, but can't imagine how long it would have taken without any hints ;)

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Can I get this reverted, it breaks unit tests for drools-wb, but the fix will take some time to get green.

**Comment by Rikkola** [comment]
Actually not sure if github has a revert button, I'll make a PR just in case.

**Comment by mareknovotny** [comment]
The Revert button is there, but it generates new PR ;)

**Comment by Rikkola** [comment]
@mareknovotny managed to find it, hehe.

**Pull request merged**